Update documentation to support InnerProduct with k-NN Lucene Engine (#6703)
opensearch-project/documentation-website: _search-plugins/knn/knn-index.md

### Supported Lucene methods

Method name | Requires training | Supported spaces | Description
:--- | :--- |:--------------------------------------------------------------------------------| :---

> `hnsw` | false | l2, cosinesimil, innerproduct (supported in OpenSearch version 2.13 and later) | Hierarchical proximity graph approach to Approximate k-NN search.

Reviewer: ```suggestion
`hnsw` | false | l2, cosinesimil, innerproduct (supported in OpenSearch 2.13 and later) | Hierarchical proximity graph approach to approximate k-NN search.
```

> `hnsw` | false | l2, cosinesimil, innerproduct (supported in OpenSearch version 2.13 and later) | Hierarchical proximity graph approach to Approximate k-NN search.

Reviewer: It looks like we are inconsistent in our capitalization of "approximate" here: https://opensearch.org/docs/latest/search-plugins/knn/approximate-knn/.

Author: @natebower fixed all the places where there was issue in the file.

Reviewer 2: Thank you, @navneet1v! Ready to merge.

> `hnsw` | false | l2, cosinesimil, innerproduct(supported with OpenSearch version 2.13 and above) | Hierarchical proximity graph approach to Approximate k-NN search.

Reviewer 2: ```suggestion
`hnsw` | false | l2, cosinesimil, innerproduct (supported in OpenSearch version 2.13 and later) | Hierarchical proximity graph approach to Approximate k-NN search.
```
